src/Microsoft.ML.TorchSharp/Utils/ImageUtils.cs

```diff
@@ -0,0 +1,221 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using static TorchSharp.torch.utils.data;
+using static TorchSharp.torch;
+using TorchSharp;
+using Microsoft.ML.Data;
+
+namespace Microsoft.ML.TorchSharp.Utils
+{
+    internal class ImageUtils
+    {
+
+        public static void Postprocess(Tensor imgBatch, Tensor classification, Tensor regression, Tensor anchors, out UInt32[] predictedLabels, out Single[] score, out Single[] boxes, double scoreThreshold = 0.05, double overlapThreshold = 0.5)
+        {
+            predictedLabels = null;
+            score = null;
+            boxes = null;
+
+            if (imgBatch is null)
+            {
+                throw new ArgumentNullException(nameof(imgBatch));
+            }
+
+            if (classification is null)
+            {
+                throw new ArgumentNullException(nameof(classification));
+            }
+
+            if (regression is null)
+            {
+                throw new ArgumentNullException(nameof(regression));
+            }
+
+            if (anchors is null)
+            {
+                throw new ArgumentNullException(nameof(anchors));
+            }
+
+            using (var postprocessScope = torch.NewDisposeScope())
+            {
+                var transformedAnchors1 = TransformBbox(anchors, regression);
+                var transformedAnchors = ClipBoxes(transformedAnchors1, imgBatch);
+
+                var finalResult = new[] { new List<Tensor>(), new List<Tensor>(), new List<Tensor>() };
+
+                for (int i = 0; i < classification.shape[2]; ++i)
+                {
+                    var scores1 = torch.squeeze(classification[.., .., i]);
+                    var scoresOverThresh = scores1 > 0.05;
```

Comment:
should use `scoreThreshold ` here?
```suggestion
                    var scoresOverThresh = scores1 > scoreThreshold ;
```

Reply:
It shouldn't make a difference because line 80 does a final check and does respect the threshold. I'm not sure if changing it here will make any difference, so I'm going to leave as is for now.